src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php

```diff
@@ -492,15 +492,23 @@ public function compileSpatialIndex(Blueprint $blueprint, Fluent $command)
      */
     protected function compileKey(Blueprint $blueprint, Fluent $command, $type)
     {
-        return sprintf('alter table %s add %s %s%s(%s)',
+        $columns = collect($command->columns)->map(function (string $column) {
+            return self::isFunctionalExpression($column) ? $column : $this->wrap($column);
+        })->implode(', ');
```

Comment:
https://github.com/laravel/framework/pull/54025#discussion_r1898675719 got outdated—not sure if you still want to change that:
```suggestion
        $columns = collect($command->columns)
            ->map(fn (string $column) => self::isFunctionalExpression($column) ? $column : $this->wrap($column))
            ->implode(', ');
```